ocrd_utils/ocrd_utils/os.py

```diff
@@ -53,4 +55,12 @@ def unzip_file_to_dir(path_to_zip, output_directory):
     z.extractall(output_directory)
     z.close()
 
-
+@contextlib.contextmanager
+def atomic_write(fpath, overwrite=False):
+    if exists(fpath):
+        mode = stat(fpath).st_mode
+    else:
+        mode =  0o664
+    with atomic_write_(fpath, overwrite=True) as f:
```

Comment:
```suggestion
    with atomic_write_(fpath, overwrite=overwrite) as f:
```

But I am not sure about `overwrite` generally. The documentation says:
> :param overwrite: If set to false, an error is raised if ``path`` exists.
> Errors are only raised after the file has been written to.  Either way,
> the operation is atomic.

I cannot see any exceptions being raised in the implementation (but perhaps `os.link` does if the target exists?). But I would consider this semantic a bad interface – after all, if I don't want to overwrite, then that case should be catched _before_ writing. Since you introduced that distinction here anyway, we could `raise` ourself...

Reply:
The difference is that with `overwrite=True`, it uses `os.rename`, otherwise `os.link` followed by `os.unlink`.

> (but perhaps os.link does if the target exists?)

Yes, it does, I just tested it.

Now I'm not sure how `os.rename` does things internally but `os.rename` is what I would use, hence the `overwrite=True`. It shouldn't be parameterizable, I see no reason to use link/unlink instead of rename.

Reply:
I see, I agree, I just don't understand why you still expose an `overwrite` kwarg then, though. (You could also simplify the usage in workspace.py.)

Reply:
> I just don't understand why you still expose an overwrite kwarg then

That's what I meant, we should not expose this. Will change that.

Reply:
f92c5a3c